tests/unit/amo/components/TestAddonReviewCard.js

```diff
@@ -576,6 +583,21 @@ describe(__filename, () => {
     expect(root.find(FlagReviewMenu)).toHaveLength(0);
   });
 
+  it('renders the flag button when not signed in', () => {
+    const root = render({ review: _setReview(fakeReview) });
+
+    expect(renderControls(root).find(FlagReviewMenu)).toHaveLength(1);
+  });
+
+  it('hides the flag button if you wrote the review', () => {
```

Comment:
When I undo your patch, this test should fail but it does not. Try to make it fail.

Here's how I undid your patch:

```diff
diff --git a/src/amo/components/AddonReviewCard/index.js b/src/amo/components/AddonReviewCard/index.js
index 13013647a..ff535434a 100644
--- a/src/amo/components/AddonReviewCard/index.js
+++ b/src/amo/components/AddonReviewCard/index.js
@@ -482,7 +482,7 @@ export class AddonReviewCardBase extends React.Component<InternalProps> {
         ) : null}
 
         {/* Do not render the "flag" button for reviews made by the current user */}
-        {flaggable && review && (!siteUser || siteUser.id !== review.userId) ? (
+        {flaggable && review ? (
           <FlagReviewMenu
             isDeveloperReply={this.isReply()}
             openerClass="AddonReviewCard-control"
```

Reply:
Ok - here's how I think we should proceed:

The current test:

* Signs the user in
* Creates a review
* Checks that the flag button isn't rendered

The ideal test:

* Signs the user in
* Creates a review
* Checks that the flag button isn't rendered
* **Signs the user out**
* **Checks that the flag button is rendered**